controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Application.java

```diff
@@ -185,8 +186,12 @@ public ApplicationActivity activity() {
     }
 
     /** Returns the global rotation of this, if present */
-    public Optional<ApplicationRotation> rotation() {
-        return rotation.map(rotation -> new ApplicationRotation(id, rotation));
+    public Optional<RotationId> rotation() {
+        return rotation;
+    }
+
+    public Optional<ApplicationRotation> applicationRotation(SystemName system) {
```

Comment:
I suggest renaming `ApplicationRotation` to something like `ApplicationDns` or `GlobalDns`. Same for method name.

Reply:
Add/update javadoc to clarify difference beteween this and `rotation()`.